include/pagmo/detail/custom_comparisons.hpp

```diff
@@ -90,7 +90,7 @@ inline bool equal_to_f(T a, T b)
 }
 
 // equal_to_vf than compares vectors of floating point types considering nan==nan
-template <typename T>
+template <typename T, detail::enable_if_is_floating_point<T> = 0>
```

Comment:
The ``enable_if`` stuff is typically used in template specialisations, if you really want to have a check here it should go as a static assert in the body of the class:

```static_assert(std::is_floating_point<T>::value, "This class can be used only with floating-point types.");```